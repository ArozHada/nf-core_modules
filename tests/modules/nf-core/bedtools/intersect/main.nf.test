nextflow_process {

    name "Test Process BEDTOOLS_INTERSECT"
    script "modules/nf-core/bedtools/intersect/main.nf"
    process "BEDTOOLS_INTERSECT"
    config "tests/modules/nf-core/bedtools/intersect/nextflow.config"
    tag "bedtools_intersect"

    test("Sarscov2 Genome BED") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                            [ id:'test' ],
                            file(params.test_data['sarscov2']['genome']['test_bed'], checkIfExists: true),
                            file(params.test_data['sarscov2']['genome']['test2_bed'], checkIfExists: true)
                           ]
                input[1] = [[:],[]]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(path(params.outdir).list()).match("bed") }
            )
        }
    }

    test("Sarscov2 Genome BED With BAM") {

        config "tests/modules/nf-core/bedtools/intersect/suffix_bam.config"

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                            [ id:'test' ],
                            file(params.test_data['sarscov2']['illumina']['test_paired_end_bam'], checkIfExists: true),
                            file(params.test_data['sarscov2']['genome']['baits_bed'], checkIfExists: true)
                           ]
                input[1] = [[:],[]]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(path(params.outdir).list()).match("bed_bam") }
            )
        }
    }
}
