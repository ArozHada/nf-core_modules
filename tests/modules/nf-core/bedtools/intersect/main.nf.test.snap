{
    "bed": {
        "content": [
            [
                "test_out.bed:md5,afcbf01c2f2013aad71dbe8e34f2c15c",
                "versions.yml:md5,b75347c6df1358fdf017dbe47c18af7f"
            ]
        ],
        "timestamp": "2023-04-11T21:19:18+0000"
    },
    "bed_bam": {
        "content": [
            [
                "test_out.bam:md5,738324efe2b1e442ceb6539a630c3fe6",
                "versions.yml:md5,b75347c6df1358fdf017dbe47c18af7f"
            ]
        ],
        "timestamp": "2023-04-11T21:19:18+0000"
    }
}